ui/admin-portal/src/app/services/candidate-field.service.ts

```diff
@@ -366,4 +399,32 @@ export class CandidateFieldService {
     return status;
   }
 
+  public getNextStep(candidate: Candidate): string {
+    let candidateOppForThisList: CandidateOpportunity = this.findRelevantCandidateOpp(candidate);
+    return candidateOppForThisList.nextStep;
+  }
+
+  public getAddedBy(candidate: Candidate): string {
+    let candidateOppForThisList: CandidateOpportunity = this.findRelevantCandidateOpp(candidate);
+    return candidateOppForThisList.createdBy.firstName + " " +
+      candidateOppForThisList.createdBy.lastName;
+  }
+
+  /**
+   * Populates the tooltip for values in the Added By column — saves table space by providing the
+   * partner name in this form.
+   * @param candidate the given candidate for this table row
+   */
+  public getAddedByPartner(candidate: Candidate): string {
+    const candidateOppForThisList: CandidateOpportunity = this.findRelevantCandidateOpp(candidate);
+    return candidateOppForThisList.createdBy.partner.name;
+  }
+
+  // When displaying a submission list, will find the relevant candidate opp for given candidate.
+  private findRelevantCandidateOpp(candidate: Candidate): CandidateOpportunity {
+    return candidate.candidateOpportunities.find(
```

Comment:
Can this return null? I imagine it can - but I don't see how the calling code handles nulls/undefined

Reply:
Can a candidate be on a submission list but not have a candidate opp for the job that the sub list relates to? Perhaps I should put it in there as a safeguard anyway but it didn't seem necessary initially.

Reply:
Null handling added per discussion